tests/test_references.py

```diff
@@ -91,35 +95,61 @@ def cb():
     assert count == 1
 
 
+def test_check_nr_arguments_of_weak_callback():
+    class Counter:
+        def cb(self, new, old, too_much):
+            pass
+
+    counter = Counter()
+    state = reactive({})
+
+    with pytest.raises(WrongNumberOfArgumentsError):
+        watch(
+            state,
+            counter.cb,
+            sync=True,
+            deep=True,
+        )
+
+
+@pytest.mark.xfail
 def test_qt_integration(qapp):
-    class Widget(QtWidgets.QWidget):
+    class Label(QtWidgets.QLabel):
         count = 0
 
         def __init__(self, state):
             super().__init__()
             self.state = state
-            self.label = QtWidgets.QLabel()
 
             self.watcher = watch(
                 self.count_display,
-                self.label.setText,
+                # Use a method from QLabel directly as callback. These methods don't
+                # have a __func__ attribute, so currently this won't be wrapped by
+                # the 'weak' wrapper in the watcher, so it will create a strong
+                # reference instead to self.
+                # Ideally we would be able to detect this and wrap it, but then
+                # there is the problem that these kinds of methods don't have a
+                # signature, so we can't detect the number of arguments to supply.
+                # I guess we can assume that we should supply only two arguments
+                # in those cases: 'self' and 'new'?
+                # We'll solve this if this becomes an actual problem :)
```

Comment:
I guess you could specifically support Qt objects by checking if the object is an instance of QObject, but that's a big can of worms for another day...